### DISCUSSION TITLE
Adding NKRO libs to an existing project

### DISCUSSION THREAD

**Discussion by SpoonyAllen** [NicoHood/HID#386] [Q&A]
Please pardon the 'n00bness'. I have a small project that I am using an arduino leonardo with. I already modified an existing project to fit my needs until i realized it was built using the standard arduino keyboard libs and i require a few more keys using nkro. I am using this to control an indie game we have developed and placed in an arcade machine. Because of the action in the game, more simultaneous inputs beyond 6 are needed. 

I have this working decently well so far. My only issue is I'm not entirely sure how to enter the keycodes properly. It seems the code is outputting most of the keys in ascii . Not a problem until i realized i need both the up arrow and F1 key. the NKRO code really doesn't give much insight on how to add these special characters and I'm sort of at a loss at this point on how i can use this to implement different keys.

I've attached my code here so anyone can see (and use it if it works for you) Any help you could give me would be greatly appreciated.

edit: pasting code here looked a little weird so i posted it to pastebin

https://pastebin.com/Ak9L76iR

**Comment by NicoHood** [comment]
Checkout the examples of the normal keyboard. The syntax should be the same

**Reply by SpoonyAllen** [reply]
Thank you for the response. I have tried reworking using some other parts of the keyboard code examples but so far i have had no luck. The lower routine of my code monitors which buttons are being pressed and so far NKRO.Keyboard.Press and .Release seem to be the only functions that work with my code. For each char i enter i get this error about reverting to ascii chars.

 `warning: narrowing conversion of '12585' from 'int' to 'uint8_t {aka unsigned char}' inside { } [-Wnarrowing] };
`
